src/adapter/src/coord.rs

```diff
@@ -1851,10 +1885,19 @@ impl Coordinator {
                 }
             };
 
+            // Track the wall time for each message for reporting.
+            let histogram_metric = self
+                .metrics
+                .slow_message_handling
+                .with_label_values(&[msg.kind()]);
+
             self.handle_message(msg)
                 // All message processing functions trace. Start a parent span for them to make
                 // it easy to find slow messages.
                 .instrument(span!(Level::DEBUG, "coordinator message processing"))
+                .wall_time()
+                .observe(histogram_metric)
+                .with_filter(move |wall_time| wall_time >= reporting_threshold)
```

Comment:
Is it a problem to do this for all messages? How much overhead would it add? It doesn't increase the prom cardinality. What about: record histogram times for all messages. But log at INFO any messages that exceeded the threshold with their full debug output? One concern, that may dump customer info to the logs. But we...probably already do that so it's presumably not worse than current practice.

Reply:
We only log for slow messages, the `with_filter(...)` call causes us to skip logging for anything faster than our reporting threshold. Otherwise, is it okay to wrap every Coordinator message in a `WallTimeFuture`? I think so, if we're concerned about this I can work to further optimize those helpers.

I'm totally down to add more logging, e.g. something along the lines of your INFO proposal. I'm wondering if we want to merge this first, get some data and/or tune the reporting threshold, and then add specific additional logging? What I want to prevent is adding some logging, running out of time to look at it, and then our logs become spammed with messages we don't need. What do you think?

Reply:
Let's merge as is. The histograms will have enough bucketed data for us to make some informed choices.

Reply:
Sounds like plan!